JDK12: set system properties java.vendor, java.vendor.url and java.vendor.url.bug

JasonFengJ9 opened the pull request:
`JDK12`: set system properties `java.vendor`, `java.vendor.url` and `java.vendor.url.bug`

`JDK12` `VersionProps.init(systemProperties)` sets system properties `java.vendor`, `java.vendor.url`, `java.vendor.url.bug`, `java.class.version`, `java.specification.version` and `java.specification.vendor`. Avoid duplication property setting in `OpenJ9`;
Also moved system properties `java.vm.specification.version` and `java.vm.vendor` setting from native to Java code to use the value of `java.specification.version` set by `VersionProps.init(systemProperties)`.

Depends on https://github.com/ibmruntimes/openj9-openjdk-jdk12/pull/13

Fixes #4660

Reviewer: @pshipton 
FYI: @DanHeidinga 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

pshipton commented:
Please add a comment to runtime/include/vendor_version.h about the new behavior in Java 12 and later. Also consider using an #ifdef to remove JAVA_VENDOR_URL from this file for Java 12 and later.

DanHeidinga commented on jcl/src/java.base/share/classes/java/lang/System.java:
```diff
@@ -378,7 +378,9 @@ private static void ensureProperties() {
 	systemProperties.put("sun.jnu.encoding", platformEncoding); //$NON-NLS-1$
 	systemProperties.put("file.encoding", fileEncoding); //$NON-NLS-1$
 	systemProperties.put("file.encoding.pkg", "sun.io"); //$NON-NLS-1$ //$NON-NLS-2$
+	/*[IF !Java12]*/
```
@JasonFengJ9 Can you confirm that this means that this is exclude for 12+?  I always get the java preprocessor tags confused :)

JasonFengJ9 replied:
@DanHeidinga yes, that's correct. 
`!Java12` means it excludes the content decorated within `pConfig` specified with JPP flag `Java12`, i.e., `12+` assuming all future Java version pConfig specify `Java12` as well.

DanHeidinga replied:
Thanks for the confirmation!

JasonFengJ9 commented:
> Please add a comment to runtime/include/vendor_version.h about the new behavior in Java 12 and later. Also consider using an #ifdef to remove JAVA_VENDOR_URL from this file for Java 12 and later.

Decorated the defines `JAVA_VENDOR` and `JAVA_VENDOR_URL` with `#if JAVA_SPEC_VERSION < 12`, added comments. In addition, moved system property setting for `java.vm.vendor` from native to Java for `JDK12` just like `java.vm.specification.version`.
Please have another look.

pshipton commented on jcl/src/java.base/share/classes/java/lang/System.java:
```diff
@@ -378,7 +378,9 @@ private static void ensureProperties() {
 	systemProperties.put("file.encoding", fileEncoding); //$NON-NLS-1$
 	systemProperties.put("file.encoding.pkg", "sun.io"); //$NON-NLS-1$ //$NON-NLS-2$
+	/*[IF !Java12]*/
 	systemProperties.put("java.specification.vendor", "Oracle Corporation"); //$NON-NLS-1$ //$NON-NLS-2$
```
I assume java.specification.vendor is set by VersionProps.init()? Please add a comment.

JasonFengJ9 replied:
That's right. Added a comment.
Ready for another review.

pshipton commented:
jenkins test all zlinux jdk11

JasonFengJ9 commented:
`travis-ci PR` build failed with `The command "bash ./buildenv/travis/build-on-travis.sh" exited with 5.` but `JDK11 extended/sanity` passed.
